appGeo/src/tests/game_tests.spec.ts

```diff
@@ -0,0 +1,231 @@
+
+    assert.equal(game1.getSnapshot(snap3.getSnapshotID()), snap3, "Can successfully get snapshot");
+
+    assert.equal(game1.removeSnapshot(snap3.getSnapshotID()), snap3, "Can successfully remove snapshot");
+    snapArray.pop();
+    assert.equal(game1.getSnapshots(), snapArray, "Snap list successfully updated");
+
+    assert.equal(game1.getSnapshot(snap3.getSnapshotID()), "", "A lack of snapshot should return failure");
+
+    */
+
+    const chat1 = new Chat(1);
+    const chat2 = new Chat(2);
+    const chat3 = new Chat(3);
+
+    const chatArray = [chat1];
+
+    assert.equal(game1.addChat(chat1), SUCCESS, "Chat added successfully");
+    assert.deepEqual(game1.getChats(), chatArray, "chat list successfully updated");
+    assert.equal(game1.getChat(chat1.getChatID()), chat1, "Can successfully get Chat");
+
+    assert.equal(game1.addChat(chat2), SUCCESS, "Chat added successfully");
+    chatArray.push(chat2);
+    assert.deepEqual(game1.getChats(), chatArray, "chat list successfully updated");
+    assert.equal(game1.getChat(chat2.getChatID()), chat2, "Can successfully get Chat");
+
+    assert.equal(game1.addChat(chat3), SUCCESS, "Chat added successfully");
+    chatArray.push(chat3);
+    assert.deepEqual(game1.getChats(), chatArray, "chat list successfully updated");
+
+    assert.equal(game1.getChat(chat3.getChatID()), chat3, "Can successfully get Chat");
+
+    assert.equal(game1.removeChat(chat3.getChatID()), chat3, "Can successfully remove Chat");
+    chatArray.pop();
+    assert.deepEqual(game1.getChats(), chatArray, "chat list successfully updated");
+
+    assert.equal(game1.getChat(chat3.getChatID()), "", "A lack of Chat should return failure");
+
+
+});
+
+QUnit.test("Game Start and End", function(assert) {
+    const now = new Date();
+    const next_week = new Date(now.getTime() + 7 * 24 * 60 * 60 * 1000);
+
+    const test_map = new CampusMap();
+
+    const Location1 = new Location(1);
+
+    const player1 = new Player(1, 'player1', Location1, ALIVE); 
+    const player2 = new Player(1, 'player2', Location1, ALIVE);
+    const player3 = new Player(1, 'player3', Location1, ALIVE);
+
+   const playerArray = [player1, player2, player3];
+
+   const playersMap = new Map()
+    for (const p of playerArray) {
+        playersMap.set(p.userID, p)
+      }
+
+    const test_game = new Game(next_week, test_map, playersMap);
+
+    assert.equal(test_game.startGame(), FAILURE, "Game cannot start with too few players");
+    assert.false(test_game.isGameActive(), "Unstarted game is not active");
+    assert.equal(test_game.endGame(),FAILURE, "Unstarted game cannot end");
+    assert.false(test_game.isGameActive(), "Unstarted game is not active");
+
+    const player4 = new Player(1, 'player4', Location1, ALIVE);
+    const player5 = new Player(1, 'player5', Location1, ALIVE);
+
+    test_game.addPlayer(player4);
+    test_game.addPlayer(player5);
+
+    assert.equal(test_game.PlayerCount, 3, "Can count players outside of game");
```

Comment:
test_game.PlayerCount? Currently not calling this as a function

Reply:
this is correct. This is how Nanci modified the function as a getter

Reply:
Ah, that is pretty confusing, wow. Okay, if that's how it is. Seems a little inconsistent since all other functions seem to always be called in form "get<>()"; but no code issue here.

Reply:
Yeah, I was gonna bring this up later as a "we should standardize some stuff" thing. It's inconsistent across game, map, player, and chat from what I saw because the **get function** vs **getFunction()** exists multiple times.

Reply:
Yeah. To be fair, I'm a personal vote of getFunction(); but we can bring this up later for sure; seems a bit out of scope for this current PR.